### ISSUE TITLE
.mgl or tapto game shortcuts freeze mister

### ISSUE THREAD

**Issue by jacob-p-vincent** [MiSTer-devel/PSX_MiSTer#306]
If you have direct video output enabled (direct_video=1) and use wizzo's favorites shortcut or tapto NFC card to launch directly into a Playstaion game, mister will freeze at the Sony screens before the game loads.  From there, you have to hard reset or pull power on mister to reset. When direct video is off (direct_video=0) in your .ini, these shortcuts work fine on every attempt.  Adding delays, etc to the the shortcut or Tapto load command sequence do not resolve this after loading games directly on a fresh mister startup.  If you load a game through the core on a fresh boot, then use Favorites or Tapto cards after that to load games directly, they work perfectly.

**Comment by birdybro** [comment]
Do you see this same behavior for Saturn, MegaCD or TGFX16-CD games?

**Comment by birdybro** [comment]
One of the main things different with this core that I could think of is it has that delay code that pops up the OSD after a counter hits a certain point. Do you have issues with the NES core like this? The NES core does something similar.

**Comment by jacob-p-vincent** [comment]
I have no issues with the NES core and using shortcuts or tapto.  There's sometimes a region popup in the OSD when launching games in the PSX core with tapto/favorites.  It's at that point that it locks up.

**Comment by silmalik** [comment]
Hi, I have exact same issue on PSX launch using MGL with direct_video=1, but I've noticed that crash only occurs when a savestate for the game exists. Once the savestate file is deleted, the game can be launched without issue.

**Comment by jacob-p-vincent** [comment]
Hmm, I only have one save state and I deleted it.  After launching a game through Tapto after a mister reboot, the issue is still there for me.

**Comment by birdybro** [comment]
Can you please attach the .mgl file you are using?

**Comment by silmalik** [comment]
One of .mgl: [Wipeout 2097 (Europe) chd.mgl.txt](https://github.com/user-attachments/files/17685015/Wipeout.2097.Europe.chd.mgl.txt)

The HDMI input of my scaler (RT4K) broke down a few days ago... I made the connections differently and it is now impossible for me to reproduce the problem... no more freeze.

Before (issue): Mister HDMI -> RT4K HDMI input
After (no issue):   Mister HDMI -> HDMI to VGA Converter -> RT4K VGA Input

**Comment by jacob-p-vincent** [comment]
I also use DV1 mode from the RetroTink4K to a 4K display.  I also use composite video to a CRT using the misteraddons yc adapter.  Only when I have direct video disabled and use the mister internal scaler over HDMI does it work correctly.

**Comment by aeyjpn** [comment]
Same as mine.
Freeze when use Tapto NFC (PN532 NFC USB Module - Type C) with PSX core and direct_video=1
When I change direct_video=0, it solved.

**Comment by TheJesusFish** [comment]
Are folks still seeing this?

I ask because, every now and again, Mister Main will crash while trying to boot a PSX game in Super Attract Mode (which uses MGLs), and I'd love to track down a way to reproduce it reliably.

**Comment by jacob-p-vincent** [comment]
Yes, the issue still persists for me.

**Comment by TheJesusFish** [comment]
Ok cool.

I've also been seeing this for a while. Do you have an MGL that triggers it 100% of the time?

**Comment by jacob-p-vincent** [comment]
For me, it's really just any PlayStation game shortcut loaded by an NFC reader right after powering on the mister. If I launch the core first and then use my NFC card to load a PlayStation game, it's fine and won't hard lock on the psx bios screen.


________________________________

**Comment by TheJesusFish** [comment]
Yeah, it freezes on the white PSX screen, right? Not the black one.

**Comment by TheJesusFish** [comment]
@jacob-p-vincent I think you're in the mister discord. I pinged you with an experimental mister main to try.

**Comment by jacob-p-vincent** [comment]
Thanks, I just saw it.  I'll give it a try soon and report back.

**Comment by jacob-p-vincent** [comment]
@TheJesusFish The mister main does resolve the issue, thanks.  If you launch with zaparoo, it will work first try on boot. However, if you remove the card, return to the menu, then try to reload the same game, it blacks screens until you remove the card. For clarity, I use a 3d printed spring loaded reader, so I leave the cards in place -I don't tap and remove.  So, me "ejecting" a card takes me back to the menu.  If I push the card back into the reader, that's when the black screen happens.  Good news is that it's not a hard lock.  I tried this with Mega Man 8 undub and SotN, both had the issue.  If you load one game on boot, remove the card, then load another different game, it appears to work as expected.  You just cant restart a game you already closed until a mister power cycle happens.

**Comment by TheJesusFish** [comment]
Hmm, I am not sure what that is about - but it sounds unrelated, no? Maybe a zaparoo issue (I also have a reader where you insert the card and let it sit).

Either way, I'm glad this other bug is finally fixed.

**Comment by TheJesusFish** [comment]
Fix is in the latest stable main.

**Comment by mlwhan2209** [comment]
amazing this is fixed, I can boot the mister then use my zaparoo card and go right into psx games. thanks so much been waiting for this
